network/transport/v2/protocol.go

```diff
@@ -71,22 +116,135 @@ func (p protocol) UnwrapMessage(envelope interface{}) interface{} {
 func (p protocol) Configure(_ transport.PeerID) {
 }
 
-func (p protocol) Start() {
+func (p *protocol) setupNatsHandler() error {
+	conn, err := p.natsConnectHandler(
+		p.config.Nats.Hostname,
+		p.config.Nats.Port,
+		time.Duration(p.config.Nats.Timeout)*time.Second,
+	)
+	if err != nil {
+		return err
+	}
+
+	js, err := conn.JetStream()
+	if err != nil {
+		return err
+	}
+
+	if _, err = js.Subscribe(events.PrivateTransactionsSubject, func(msg *nats.Msg) {
+		if err := p.handlePrivateTx(msg); err != nil {
+			logrus.Errorf("failed to handle private transaction: %v", err)
+
+			if err := msg.Nak(); err != nil {
+				logrus.Errorf("failed to NACK private transaction event: %v", err)
+			}
+
+			return
+		}
+
+		if err := msg.Ack(); err != nil {
+			logrus.Errorf("failed to ACK private transaction event: %v", err)
+		}
+	}); err != nil {
+		conn.Close()
+		return err
+	}
+
+	go func() {
+		<-p.ctx.Done()
+		conn.Close()
+	}()
+
+	return nil
+}
+
+func (p *protocol) Start() {
+	p.ctx, p.cancel = context.WithCancel(context.Background())
+
+	go func() {
+		for {
+			if err := p.setupNatsHandler(); err != nil {
+				logrus.Errorf("failed to setup NATS handler: %v", err)
+
+				select {
+				case <-p.ctx.Done():
+					return
+				case <-time.After(5 * time.Second):
+					continue
+				}
+			}
+
+			return
+		}
+	}()
 }
 
-func (p protocol) Stop() {
+func (p *protocol) handlePrivateTx(msg *nats.Msg) error {
+	if p.nodeDID == nil {
+		return errors.New("node DID is not set")
```

Comment:
Although the business layer will prevent private TXs without a nodeDID, the logic in the network here will create an infinite loop. This error will NACK the event.

Reply:
Good one. This is awful because it will just exhaust CPU by constantly NAKking messages if the Node DID isn't configured. I'll check for the node DID while setting up the NATS subscriber and error there.

Reply:
This is not really an option as it's no longer fixed but uses an interface so it could change at runtime (in theory)

Reply:
Let's fix it when implementing retry